tests/testutils/kubeutils/kind_cluster.go

```diff
@@ -152,6 +162,54 @@ func (k KindCluster) Apply(manifests string) (stdOut, stdErr bytes.Buffer, err e
 	return k.runKubectl(stdin, "apply", "-f", f.Name())
 }
 
+func (k KindCluster) WaitForPods(podNameRegex, namespaceName string, duration time.Duration) []*corev1.Pod {
+	namePattern, e := regexp.Compile(podNameRegex)
+	require.NoError(k.Testcase, e)
+
+	var running []*corev1.Pod
+	require.Eventually(k.Testcase, func() bool {
+		ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+		defer cancel()
+		pods, err := k.Clientset.CoreV1().Pods(namespaceName).List(ctx, metav1.ListOptions{})
+		require.NoError(k.Testcase, err)
+		cancel()
+
+		matches := map[string]struct{}{}
+
+		for i := range pods.Items {
+			pod := pods.Items[i]
+			if namePattern.MatchString(pod.Name) {
+				matches[pod.Name] = struct{}{}
+			}
+		}
+
+		if len(matches) == 0 {
+			return false
+		}
+
+		var matched []*corev1.Pod
+		for name := range matches {
+			ctx, cancel = context.WithTimeout(context.Background(), 5*time.Second)
+			pod, err := k.Clientset.CoreV1().Pods(namespaceName).Get(ctx, name, metav1.GetOptions{})
+			require.NoError(k.Testcase, err)
+			cancel()
+			k.Testcase.Logger.Debug(fmt.Sprintf("%s is: %s", pod.Name, pod.Status.Phase))
+			if pod.Status.Phase == corev1.PodRunning {
+				delete(matches, name)
+				matched = append(matched, pod)
+			}
+		}
+
+		if len(matches) == 0 {
+			running = matched
+			return true
+		}
+		return false
+	}, duration, time.Second)
```

Comment:
I guess we could check that duration is greater than a second if we fix polling to a second here. Not a big deal, but line 171 also sets 5 seconds for the timeout, should it be changed to match the poll interval?

Reply:
Sorry, what's the issue here?

Reply:
I think if duration is less than a second, then polling never runs. The context is set to 5s, so we are running an operation with a timeout of 5s with a polling period of one second.

Reply:
I think the implementation of the testify Eventually asserter ensures there's only one invocation at a time and its actual interval is the max of the ticker and call durations.

If someone expects a timeout duration for less than a second I think the resulting assertion failure would be helpful enough to communicate that their expectations don't meet the execution (so the duration would have to be bumped up to give k8s time to get the pod running).